latch/registry/table.py

```diff
@@ -541,6 +560,119 @@ def _add_record_deletes_selection(
 
         mutations.append(res)
 
+    # upsert column
+
+    def create_column(
+        self,
+        key: str,
+        type: Union[Type[RegistryPrimitivePythonValue], str, List[str]],
+        *,
+        allow_multiple_entries: bool = False,
+        required: bool = False,
+    ):
+        """Create a column.
+
+        A transport error is thrown if a column of the same name already exists.
+
+        Args:
+            name: New column name.
+            type: Type of new column. If a string is provided, it is treated
+                as the id of Table and will produce a link column. If a list of
+                strings is provided, they are treated as members of an Enum and
+                will produce a select column.
+            allow_multiple: Whether or not to allow multiple entries in the
+                new column. Only valid if `type` is a file, directory, or link.
+            required: Whether or not a value is required in the new column.
+
+        """
+
+        if allow_multiple_entries:
+            if type in {str, int, float, date, datetime, bool}:
+                raise InvalidColumnError(
+                    key, str(type), f"{str(type)} columns cannot have multiple entries"
+                )
+            if isinstance(type, List):
+                raise InvalidColumnError(
+                    key, "enum", "enum columns cannot have multiple entries"
+                )
+
+        registry_type: Optional[RegistryType] = None
+        if type is str:
+            registry_type = {"primitive": "string"}
+        if type is int:
+            registry_type = {"primitive": "integer"}
+        if type is float:
+            registry_type = {"primitive": "number"}
+        if type is date:
+            registry_type = {"primitive": "date"}
+        if type is datetime:
+            registry_type = {"primitive": "datetime"}
+        if type is bool:
+            registry_type = {"primitive": "boolean"}
+        if type is LatchFile:
+            registry_type = {"primitive": "blob"}
+        if type is LatchDir:
+            registry_type = {"primitive": "blob", "metadata": {"nodeType": "dir"}}
+
+        if isinstance(type, str):
+            registry_type = {"primitive": "link", "experimentId": type}
+        if isinstance(type, List):
+            if all(isinstance(t, str) for t in type):
+                registry_type = {"primitive": "enum", "members": type}
+
+        if registry_type is None:
+            raise InvalidColumnError(
+                key,
+                str(type),
+                f"python type {type} could not be converted to registry type",
+            )
+
+        if allow_multiple_entries:
+            registry_type = {"array": registry_type}
```

Comment:
I would move the allow_multiple_entries check here and check the constructed type.